### PR TITLE
feat(client): Allow disabling environment variable override in Java client

### LINKED ISSUE #9401: Allows disabling environment variable override in JavaClient
**Is your feature request related to a problem? Please describe.**

In the Java Client there is a hard overwrite of configuration properties for any client: https://github.com/camunda/zeebe/blob/87bd591e6b3aaea99cc87f818db4c8e0bfcc72cb/clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientBuilderImpl.java#L297

This can cause problem if the client is configured in a scenario, where it is absolutely not expected, for example

1. Test cases. If I create the zeebe-process-test engine and obtain a ZeebeClient from it, I expect this to connect to that test engine - and not take any kind of environment variables into account. A great example of the confusion it causes can be found here: https://github.com/camunda-community-hub/spring-zeebe/issues/199

2. Spring: Spring has its own mechanism to use environment variables for configuration. It is confusing for Spring users that there is a second possibility that might even overwrite the Spring config.

**Describe the solution you'd like**

I think we can't easily disable the functionality completely because of backward compatibility.

Hence, I would add a **feature toggle** in the client to **disable environment variable overrides**.

### PR DESCRIPTION
**Pull request by berndruecker**
This would fix https://github.com/camunda/zeebe/issues/9401, which is needed to reliably run test cases using zeebe-process-test or spring-zeebe.

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientBuilderImpl.java [hunk 1]
```diff
@@ -294,8 +307,10 @@ public ZeebeClientBuilder overrideAuthority(final String authority) {
 
   @Override
   public ZeebeClient build() {
-    applyOverrides();
-    applyDefaults();
+    if (applyEnvironmentVariableOverrides) {
+      applyOverrides();
+      applyDefaults();
```
Comment (npepinpe) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientBuilderImpl.java:312]: :x: It seems `applyDefaults` is actually just setting up the OAuth credentials provider if one of the environment variables is set. Can we just inline `applyDefaults` into `applyOverrides`? I don't want someone adding things later in `applyDefaults` thinking this is always going to be executed, and have a flag which disable environment overrides somehow turn that off.

**Review by npepinpe — APPROVED** [review]
:+1: 

I had just one small comment, but I don't think we need another review for that. Please address it and then go ahead and merge :rocket:

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientBuilderImpl.java (hunk 1 above)
Comment (berndruecker) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientBuilderImpl.java]: Makes total sense to me! I also invested some time to figure out if applyDefaults needs to run or not :-) Added

**Comment by berndruecker** [comment]
I was surprised not to see this in 8.0.3 - was there any reason not to include it (I would love to push it into spring-zeebe asap :-)) - maybe @npepinpe has an idea?

**Comment by npepinpe** [comment]
No, sorry, we just forgot to backport it. Normally we label PRs to be backported, so merging them causes it to be backported immediately, and I forgot to do it here.

**Comment by Zelldon** [comment]
Wow that was intense :sweat_smile:

**Comment by npepinpe** [comment]
Whoa, not sure what's going on with the bot :smile: @korthout ?

**Comment by npepinpe** [comment]
I appreciate the enthusiasm though :smile:

**Comment by npepinpe** [comment]
How do we make it stop? :smile:

**Comment by berndruecker** [comment]
Wow - a lot of activity in this issue :-) Does that mean this should be finally in 8.0.4 - right?

**Pull request merged**